R/parse_depends.R

```diff
@@ -16,83 +20,119 @@ format_depend <- function(package, version, remote_provider){
+  # CRAN canonical URL
+  urls <- c(
+    "Comprehensive R Archive Network (CRAN)" =
+      "https://CRAN.R-project.org/package=%s",
+    "BioConductor" =
+      "https://bioconductor.org/packages/release/bioc/html/%s.html",
+    "_GITHUB_" =
+      "https://github.com/%s"
+  )
+
+  # The remote provider takes precedence over the non-remote provider
+  if (remote_provider != "") {
 
+    sprintf(urls["_GITHUB_"], stringr::str_remove(remote_provider, "github::"))
+
+  } else if (! is.null(provider) && provider$name %in% names(urls)) {
+
+    sprintf(urls[provider$name], identifier)
+
+  } # else NULL implicitly
 }
 
+
 parse_depends <- function(deps) {
 
-  purrr::pmap(list(deps$package, deps$version,
-                deps$remote_provider),
-  format_depend)
+  purrr::pmap(
+    list(deps$package, deps$version, deps$remote_provider),
+    format_depend
+  )
 }
 
 
-## FIXME these are not version-specific. That's often not accurate, though does reflect the CRAN assumption that you must be compatible with the latest version...
+## FIXME these are not version-specific. That's often not accurate, though does
+## reflect the CRAN assumption that you must be compatible with the latest
+## version...
 guess_dep_id <- function(dep) {
+
   if (dep$name == "R") {
+
     ## FIXME No good identifier for R, particularly none for specific version
     id <- "https://www.r-project.org"
+
   } else if (is.null(dep$provider)) {
+
     id <- NULL
+
   } else if (grepl("cran.r-project.org", dep$provider$url)) {
+
     id <- paste0(dep$provider$url, "/web/packages/", dep$identifier)
+
   } else if (grepl("www.bioconductor.org", dep$provider$url)) {
-    id <-
-      paste0(dep$provider$url,
-             "/packages/release/bioc/html/",
-             dep$identifier,
-             ".html")
+
+    id <- paste0(
+      dep$provider$url, "/packages/release/bioc/html/", dep$identifier, ".html"
+    )
+
   } else {
+
     id <- NULL
   }
 
   id
-
 }
 
-add_remote_to_dep <- function(package, remotes){
-  remote_provider <- remotes[grepl(paste0("/", package, "$"),
-                                   remotes)]
-  if(length(remote_provider) == 0){
-    ""
-  }else{
-    remote_provider
-  }
+
+add_remote_to_dep <- function(package, remotes) {
+
+  remote_providers <- grep(paste0("/", package, "$"), remotes, value = TRUE)
+
+  if (length(remote_providers)) remote_providers else ""
 }
 
 
 # helper to get system dependencies
-get_sys_links <- function(pkg, description = ""){
-  out1 <- jsonlite::fromJSON(sprintf("https://sysreqs.r-hub.io/pkg/%s", pkg), simplifyVector = FALSE)
-  out2 <- jsonlite::fromJSON(sprintf("https://sysreqs.r-hub.io/map/%s", curl::curl_escape(description)), simplifyVector = FALSE)
-  sysreqs <- unique(c(sapply(out1, names), sapply(out2, names)))
-  sprintf("https://sysreqs.r-hub.io/get/%s", sysreqs)
+get_sys_links <- function(pkg, description = "") {
+
+  # Define helper functions
+  to_url <- function(a, b) sprintf("https://sysreqs.r-hub.io/%s/%s", a, b)
```

Comment:
can you define the two helpers functions (`to_url()`, `get_json_names()` outside of this function and preferentially with `{}`?

Reply:
Good idea. I renamed the functions to make clear that they relate to "r-hub":

* `to_url()` -> `to_rhub_url()`
* `get_json_names()` -> `get_rhub_json_names()`

Reply:
Awesome. Now let's just wait for Travis CI to run (lot of builds in the ropensci account at the moment, I think)!